src/aideck/exceptions.py

```
@@ -0,0 +1,129 @@
+        panel = Panel(content, title="[bold yellow]🔑 API Key Required[/bold yellow]", border_style="yellow", expand=False)
+        console.print(panel)
+
+
+class ConfigurationError(AideckError):
+    """
+    Exception raised when there's a configuration issue (e.g., missing Azure endpoints).
+    
+    This exception automatically displays a formatted error panel.
+    """
+    
+    def __init__(self, error_type: str, detail: str, error_code: int = None):
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)